Source/SwiftLintFramework/Rules/ExplictSelfRule.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  ExplictSelfRule.swift
+//  SwiftLint
+//
+//  Created by Ian Keen on 2016-10-06.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct ExplicitSelfRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.Error)
+
+    public init() { }
+
+    public static let description = RuleDescription(
+        identifier: "explicitSelf",
+        name: "Explict Self",
+        description: "Require explicit self for instance members",
+        nonTriggeringExamples: [
+            "class Good1 { let value: Int = 42; var property: Int { return self.value } }",
+            "class Good2 { let value: Int = 42; func function() -> Int { return self.value } }",
+            "class Good3 { let value: Int; init() { self.value = 42 } }",
+        ],
+        triggeringExamples: [
+            "class Bad1 { let value: Int = 42; var property: Int { return value } }",
+            "class Bad2 { let value: Int = 42; func function() -> Int { return value } }",
+            "class Bad3 { let value: Int; init() { value = 42 } }",
+        ]
+    )
+
+    public func validateFile(
+        file: File,
+        kind: SwiftDeclarationKind,
+        dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+
+        //only check kinds that might use instance members
+        let types: [SwiftDeclarationKind] = [
+            .FunctionMethodInstance,
+            .VarInstance,
+        ]
+        guard types.contains(kind) else { return [] }
+
+        //build a set of instance memebers
+        let memberTypes: [SwiftDeclarationKind] = [
+            .FunctionMethodInstance,
+            .VarInstance,
+        ]
+        let instanceMembers = file.members(memberTypes)
+        let instanceMemberNames = instanceMembers.flatMap { $0["key.name"] as? String }
+
+        //get the members body (functions and calculated properties will have this data)
+        guard
+            let bodyLocation = (dictionary["key.bodyoffset"] as? Int64).flatMap({ Int($0) }),
```

Comment:
🤔 `as?` seems to be the convention throughout - I think it would be preferential to a crash anyway